src/overlays/actors/ovl_En_Kakasi/z_en_kakasi.h

```diff
@@ -6,14 +6,47 @@
 struct EnKakasi;
 
 typedef void (*EnKakasiActionFunc)(struct EnKakasi*, GlobalContext*);
+typedef void (*EnKakasiUnkFunc)(GlobalContext*, struct EnKakasi*);
 
 typedef struct EnKakasi {
-    /* 0x0000 */ Actor actor;
-    /* 0x0144 */ char unk_144[0x4];
-    /* 0x0148 */ EnKakasiActionFunc actionFunc;
-    /* 0x014C */ char unk_14C[0x154];
+    /* 0x000 */ Actor actor;
+    /* 0x144 */ EnKakasiUnkFunc unkFunc; // used for one z_snap calling function?
+    /* 0x148 */ EnKakasiActionFunc actionFunc;
+    /* 0x14C */ SkelAnime skelanime;
+    /* 0x190 */ s16 unk190; // camera index for song teaching angles?
+    /* 0x192 */ s16 postTeachTimer;
+    /* 0x194 */ s16 aboveGroundStatus; // from params, changed to 2 in init
+    /* 0x196 */ s16 unkState196;
+    /* 0x19A */ UNK_TYPE1 pad198[0x8];
+    /* 0x1A0 */ s32 animeIndex;
```

Comment:
```suggestion
    /* 0x1A0 */ s32 animIndex;
```
You'll want to change all uses of this too